src/basic/prioq.h

```diff
@@ -19,8 +19,14 @@ int prioq_put(Prioq *q, void *data, unsigned *idx);
 int prioq_remove(Prioq *q, void *data, unsigned *idx);
 int prioq_reshuffle(Prioq *q, void *data, unsigned *idx);
 
-void *prioq_peek(Prioq *q) _pure_;
+void *prioq_peek_item(Prioq *q, unsigned idx) _pure_;
+_pure_ static inline void *prioq_peek(Prioq *q) {
```

Comment:
strictly speaking _pure_ is probably redundant on static inline functions, no? the compiler really should be able to figure this out on its own